trinity/extensibility/plugin_manager.py

```diff
@@ -74,7 +113,55 @@ def broadcast(self, event: BaseEvent, exclude: BasePlugin = None) -> None:
             if not plugin.should_start():
                 continue
 
-            plugin.start(None)
+            plugin._start()
             self._started_plugins.append(plugin)
             self._logger.info("Plugin started: {}".format(plugin.name))
             self.broadcast(PluginStartedEvent(plugin), plugin)
+
+    def prepare(self,
+                args: Namespace,
+                chain_config: ChainConfig,
+                boot_kwargs: Dict[str, Any] = None) -> None:
+        for plugin in self._plugin_store:
+
+            if not self._is_responsible_for_plugin(plugin):
+                continue
+
+            context = self._create_context_for_plugin(plugin)
+            context.args = args
+            context.boot_kwargs = boot_kwargs
+            context.chain_config = chain_config
```

Comment:
Maybe roll this into `_create_context_for_plugin` with `**kwargs`?

Reply:
Not a big **kwargs fan because of the lower type safety. Would explicit parameter passing be ok with you?